### PR TITLE
Replace webpack html plugins with lodash template javascript

### PR DESCRIPTION
**Pull request by li-kai**
## Context
Upgrade to webpack 5 deprecates some packages. Instead of moving to other packages, why not just do the processing ourselves?

## Implementation

Use the all powerful, turing complete JavaScript, instead of configurations.

## Other Information

Here's the diff: https://www.diffchecker.com/3PIIOU1z

### Old:

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta
      name="viewport"
      content="width=device-width,user-scalable=no,initial-scale=1,viewport-fit=cover"
    />
    <link href="/apple-icon-180x180.png" rel="apple-touch-icon" sizes="180x180" />
    <link href="/favicon-32x32.png" rel="icon" type="image/png" sizes="32x32" />
    <link href="/favicon-16x16.png" rel="icon" type="image/png" sizes="16x16" />
    <link href="/safari-pinned-tab.svg" rel="mask-icon" color="#ff5138" />
    <link href="/manifest.json" rel="manifest" />
    <link
      href="https://nusmods.com/api/v2/2020-2021/moduleList.json"
      rel="preload"
      as="fetch"
      crossorigin="anonymous"
    />
    <link
      href="https://nusmods.com/api/v2/2020-2021/semesters/1/venueInformation.json"
      rel="prefetch"
    />
    <title>NUSMods</title>
    <meta
      name="description"
      content="NUSMods is a timetable builder and knowledge platform, providing students with a better way to plan their school timetable and useful module-related information that are community-driven."
      data-react-helmet="true"
    />
    <meta name="og:title" content="NUSMods" data-react-helmet="true" />
    <meta
      name="og:description"
      content="NUSMods is a timetable builder and knowledge platform, providing students with a better way to plan their school timetable and useful module-related information that are community-driven."
      data-react-helmet="true"
    />
    <meta name="og:site_name" content="NUSMods" />
    <meta name="og:type" content="website" />
    <link
      href="/opensearch.xml"
      rel="search"
      title="NUSMods Modules"
      type="application/opensearchdescription+xml"
    />
    <meta name="theme-color" content="#ffffff" />
    <link href="/app.03153e49.css" rel="stylesheet" />
    <link rel="preload" href="/304.d691046d.js" as="script" />
    <link rel="preload" href="/app.4accc938.js" as="script" />
  </head>
  <body>
    <div id="app"></div>
    <noscript>Please enable JavaScript to use NUSMods</noscript>
    <script src="/browser-warning.1b1759e2defd968e0b98.js" async defer="defer"></script>
    <script>
      !(function () {
        'use strict';
        var e,
          t,
          n,
          r = {},
          o = {};
        function u(e) {
          if (o[e]) return o[e].exports;
          var t = (o[e] = { id: e, loaded: !1, exports: {} });
          return r[e].call(t.exports, t, t.exports, u), (t.loaded = !0), t.exports;
        }
        (u.m = r),
          (u.n = function (e) {
            var t =
              e && e.__esModule
                ? function () {
                    return e.default;
                  }
                : function () {
                    return e;
                  };
            return u.d(t, { a: t }), t;
          }),
          (u.t = function (e, t) {
            if ((1 & t && (e = this(e)), 8 & t)) return e;
            if (4 & t && 'object' == typeof e && e && e.__esModule) return e;
            var n = Object.create(null);
            u.r(n);
            var r = {};
            if (2 & t && 'object' == typeof e && e)
              for (var o in e)
                r[o] = function (t) {
                  return e[t];
                }.bind(null, o);
            return (
              (r.default = function () {
                return e;
              }),
              u.d(n, r),
              n
            );
          }),
          (u.d = function (e, t) {
            for (var n in t)
              u.o(t, n) && !u.o(e, n) && Object.defineProperty(e, n, { enumerable: !0, get: t[n] });
          }),
          (u.f = {}),
          (u.e = function (e) {
            return Promise.all(
              Object.keys(u.f).reduce(function (t, n) {
                return u.f[n](e, t), t;
              }, []),
            );
          }),
          (u.u = function (e) {
            return (
              ({
                76: 'module-finder',
                121: 'planner',
                157: 'export',
                261: 'module',
                302: 'venueData',
                365: 'venue',
                481: 'today',
                536: 'contribute',
                874: 'tetris',
              }[e] || e) +
              '.' +
              {
                76: '7487e6ee',
                121: 'f91b3b3e',
                157: 'ee2eafc1',
                261: 'ae71175a',
                302: '1d4f2919',
                365: 'd7d0a04c',
                481: '6b66c212',
                536: '0603116f',
                685: 'b75ee180',
                698: '4f811851',
                709: '4f3ea2e6',
                817: '6f7cc4a1',
                874: '845db6d8',
                914: '1c0dafc7',
              }[e] +
              '.js'
            );
          }),
          (u.miniCssF = function (e) {
            return (
              ({
                76: 'module-finder',
                121: 'planner',
                143: 'app',
                157: 'export',
                261: 'module',
                302: 'venueData',
                365: 'venue',
                481: 'today',
                536: 'contribute',
                666: 'runtime',
                874: 'tetris',
              }[e] || e) +
              '.' +
              {
                76: '9cd7f4e3',
                121: '68600187',
                143: '03153e49',
                157: '31d6cfe0',
                261: '882c01d5',
                302: '31d6cfe0',
                304: '31d6cfe0',
                365: '90998fd9',
                481: 'cb258671',
                536: '320bc21d',
                666: 'undefine',
                685: '31d6cfe0',
                698: '31d6cfe0',
                709: '31d6cfe0',
                817: '31d6cfe0',
                874: '8496acf2',
                914: '31d6cfe0',
              }[e] +
              '.css'
            );
          }),
          (u.g = (function () {
            if ('object' == typeof globalThis) return globalThis;
            try {
              return this || new Function('return this')();
            } catch (e) {
              if ('object' == typeof window) return window;
            }
          })()),
          (u.hmd = function (e) {
            return (
              (e = Object.create(e)).children || (e.children = []),
              Object.defineProperty(e, 'exports', {
                enumerable: !0,
                set: function () {
                  throw new Error(
                    'ES Modules may not assign module.exports or exports.*, Use ESM export syntax, instead: ' +
                      e.id,
                  );
                },
              }),
              e
            );
          }),
          (u.o = function (e, t) {
            return Object.prototype.hasOwnProperty.call(e, t);
          }),
          (e = {}),
          (u.l = function (t, n, r) {
            if (e[t]) e[t].push(n);
            else {
              var o, i;
              if (void 0 !== r)
                for (var f = document.getElementsByTagName('script'), a = 0; a < f.length; a++) {
                  var c = f[a];
                  if (c.getAttribute('src') == t) {
                    o = c;
                    break;
                  }
                }
              o ||
                ((i = !0),
                ((o = document.createElement('script')).charset = 'utf-8'),
                (o.timeout = 120),
                u.nc && o.setAttribute('nonce', u.nc),
                (o.src = t)),
                (e[t] = [n]);
              var d = function (n, r) {
                  (o.onerror = o.onload = null), clearTimeout(l);
                  var u = e[t];
                  if (
                    (delete e[t],
                    o.parentNode && o.parentNode.removeChild(o),
                    u &&
                      u.forEach(function (e) {
                        return e(r);
                      }),
                    n)
                  )
                    return n(r);
                },
                l = setTimeout(d.bind(null, void 0, { type: 'timeout', target: o }), 12e4);
              (o.onerror = d.bind(null, o.onerror)),
                (o.onload = d.bind(null, o.onload)),
                i && document.head.appendChild(o);
            }
          }),
          (u.r = function (e) {
            'undefined' != typeof Symbol &&
              Symbol.toStringTag &&
              Object.defineProperty(e, Symbol.toStringTag, { value: 'Module' }),
              Object.defineProperty(e, '__esModule', { value: !0 });
          }),
          (u.nmd = function (e) {
            return (e.paths = []), e.children || (e.children = []), e;
          }),
          (u.p = '/'),
          (t = function (e) {
            return new Promise(function (t, n) {
              var r = u.miniCssF(e),
                o = u.p + r;
              if (
                (function (e, t) {
                  for (var n = document.getElementsByTagName('link'), r = 0; r < n.length; r++) {
                    var o = (i = n[r]).getAttribute('data-href') || i.getAttribute('href');
                    if ('stylesheet' === i.rel && (o === e || o === t)) return i;
                  }
                  var u = document.getElementsByTagName('style');
                  for (r = 0; r < u.length; r++) {
                    var i;
                    if ((o = (i = u[r]).getAttribute('data-href')) === e || o === t) return i;
                  }
                })(r, o)
              )
                return t();
              !(function (e, t, n, r) {
                var o = document.createElement('link');
                (o.rel = 'stylesheet'),
                  (o.type = 'text/css'),
                  (o.onload = n),
                  (o.onerror = function (n) {
                    var u = (n && n.target && n.target.href) || t,
                      i = new Error('Loading CSS chunk ' + e + ' failed.\n(' + u + ')');
                    (i.code = 'CSS_CHUNK_LOAD_FAILED'),
                      (i.request = u),
                      o.parentNode.removeChild(o),
                      r(i);
                  }),
                  (o.href = t),
                  document.head.appendChild(o);
              })(e, o, t, n);
            });
          }),
          (n = { 666: 0 }),
          (u.f.miniCss = function (e, r) {
            n[e]
              ? r.push(n[e])
              : 0 !== n[e] &&
                { 76: 1, 121: 1, 261: 1, 365: 1, 481: 1, 536: 1, 874: 1 }[e] &&
                r.push(
                  (n[e] = t(e).then(
                    function () {
                      n[e] = 0;
                    },
                    function (t) {
                      throw (delete n[e], t);
                    },
                  )),
                );
          }),
          (function () {
            var e = { 666: 0 },
              t = [];
            u.f.j = function (t, n) {
              var r = u.o(e, t) ? e[t] : void 0;
              if (0 !== r)
                if (r) n.push(r[2]);
                else {
                  var o = new Promise(function (n, o) {
                    r = e[t] = [n, o];
                  });
                  n.push((r[2] = o));
                  var i = u.p + u.u(t),
                    f = new Error();
                  u.l(
                    i,
                    function (n) {
                      if (u.o(e, t) && (0 !== (r = e[t]) && (e[t] = void 0), r)) {
                        var o = n && ('load' === n.type ? 'missing' : n.type),
                          i = n && n.target && n.target.src;
                        (f.message = 'Loading chunk ' + t + ' failed.\n(' + o + ': ' + i + ')'),
                          (f.name = 'ChunkLoadError'),
                          (f.type = o),
                          (f.request = i),
                          r[1](f);
                      }
                    },
                    'chunk-' + t,
                  );
                }
            };
            var n = function () {};
            function r() {
              for (var n, r = 0; r < t.length; r++) {
                for (var o = t[r], i = !0, f = 1; f < o.length; f++) {
                  var a = o[f];
                  0 !== e[a] && (i = !1);
                }
                i && (t.splice(r--, 1), (n = u((u.s = o[0]))));
              }
              return 0 === t.length && (u.x(), (u.x = function () {})), n;
            }
            u.x = function () {
              (u.x = function () {}), (i = i.slice());
              for (var e = 0; e < i.length; e++) o(i[e]);
              return (n = r)();
            };
            var o = function (r) {
                for (var o, i, [a, c, d, l] = r, s = 0, p = []; s < a.length; s++)
                  (i = a[s]), u.o(e, i) && e[i] && p.push(e[i][0]), (e[i] = 0);
                for (o in c) u.o(c, o) && (u.m[o] = c[o]);
                for (d && d(u), f(r); p.length; ) p.shift()();
                return l && t.push.apply(t, l), n();
              },
              i = (self.webpackChunk = self.webpackChunk || []),
              f = i.push.bind(i);
            i.push = o;
          })(),
          u.x();
      })();
      //# sourceMappingURL=runtime.1d37bf73.js.map
    </script>
    <script src="/304.d691046d.js"></script>
    <script src="/app.4accc938.js"></script>
  </body>
</html>

```
### New
```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta
      name="viewport"
      content="width=device-width,user-scalable=no,initial-scale=1,viewport-fit=cover"
    />
    <link href="/apple-icon-180x180.png" rel="apple-touch-icon" sizes="180x180" />
    <link href="/favicon-32x32.png" rel="icon" type="image/png" sizes="32x32" />
    <link href="/favicon-16x16.png" rel="icon" type="image/png" sizes="16x16" />
    <link href="/safari-pinned-tab.svg" rel="mask-icon" color="#ff5138" />
    <link href="/manifest.json" rel="manifest" />
    <link
      href="https://nusmods.com/api/v2/2020-2021/moduleList.json"
      rel="preload"
      as="fetch"
      crossorigin="anonymous"
    />
    <link
      href="https://nusmods.com/api/v2/2020-2021/semesters/1/venueInformation.json"
      rel="prefetch"
    />
    <link href="/304.d691046d.js" rel="preload" as="script" />
    <link href="/app.5ebfa832.js" rel="preload" as="script" />
    <link href="/app.03153e49.css" rel="stylesheet" />
    <title>NUSMods</title>
    <meta
      name="description"
      content="NUSMods is a timetable builder and knowledge platform, providing students with a better way to plan their school timetable and useful module-related information that are community-driven."
      data-react-helmet="true"
    />
    <meta name="og:title" content="NUSMods" data-react-helmet="true" />
    <meta
      name="og:description"
      content="NUSMods is a timetable builder and knowledge platform, providing students with a better way to plan their school timetable and useful module-related information that are community-driven."
      data-react-helmet="true"
    />
    <meta name="og:site_name" content="NUSMods" />
    <meta name="og:type" content="website" />
    <link
      href="/opensearch.xml"
      rel="search"
      title="NUSMods Modules"
      type="application/opensearchdescription+xml"
    />
    <meta name="theme-color" content="#ffffff" />
  </head>
  <body>
    <div id="app"></div>
    <noscript>Please enable JavaScript to use NUSMods</noscript>
    <script src="/browser-warning.1b1759e2defd968e0b98.js" async defer="defer"></script>
    <script>
      !(function () {
        'use strict';
        var e,
          t,
          n,
          r = {},
          o = {};
        function u(e) {
          if (o[e]) return o[e].exports;
          var t = (o[e] = { id: e, loaded: !1, exports: {} });
          return r[e].call(t.exports, t, t.exports, u), (t.loaded = !0), t.exports;
        }
        (u.m = r),
          (u.n = function (e) {
            var t =
              e && e.__esModule
                ? function () {
                    return e.default;
                  }
                : function () {
                    return e;
                  };
            return u.d(t, { a: t }), t;
          }),
          (u.t = function (e, t) {
            if ((1 & t && (e = this(e)), 8 & t)) return e;
            if (4 & t && 'object' == typeof e && e && e.__esModule) return e;
            var n = Object.create(null);
            u.r(n);
            var r = {};
            if (2 & t && 'object' == typeof e && e)
              for (var o in e)
                r[o] = function (t) {
                  return e[t];
                }.bind(null, o);
            return (
              (r.default = function () {
                return e;
              }),
              u.d(n, r),
              n
            );
          }),
          (u.d = function (e, t) {
            for (var n in t)
              u.o(t, n) && !u.o(e, n) && Object.defineProperty(e, n, { enumerable: !0, get: t[n] });
          }),
          (u.f = {}),
          (u.e = function (e) {
            return Promise.all(
              Object.keys(u.f).reduce(function (t, n) {
                return u.f[n](e, t), t;
              }, []),
            );
          }),
          (u.u = function (e) {
            return (
              ({
                76: 'module-finder',
                121: 'planner',
                157: 'export',
                261: 'module',
                302: 'venueData',
                365: 'venue',
                481: 'today',
                536: 'contribute',
                874: 'tetris',
              }[e] || e) +
              '.' +
              {
                76: '7487e6ee',
                121: 'f91b3b3e',
                157: 'ee2eafc1',
                261: 'ae71175a',
                302: '1d4f2919',
                365: 'd7d0a04c',
                481: '6b66c212',
                536: '0603116f',
                685: 'b75ee180',
                698: '4f811851',
                709: '4f3ea2e6',
                817: '6f7cc4a1',
                874: '845db6d8',
                914: '1c0dafc7',
              }[e] +
              '.js'
            );
          }),
          (u.miniCssF = function (e) {
            return (
              ({
                76: 'module-finder',
                121: 'planner',
                143: 'app',
                157: 'export',
                261: 'module',
                302: 'venueData',
                365: 'venue',
                481: 'today',
                536: 'contribute',
                666: 'runtime',
                874: 'tetris',
              }[e] || e) +
              '.' +
              {
                76: '9cd7f4e3',
                121: '68600187',
                143: '03153e49',
                157: '31d6cfe0',
                261: '882c01d5',
                302: '31d6cfe0',
                304: '31d6cfe0',
                365: '90998fd9',
                481: 'cb258671',
                536: '320bc21d',
                666: 'undefine',
                685: '31d6cfe0',
                698: '31d6cfe0',
                709: '31d6cfe0',
                817: '31d6cfe0',
                874: '8496acf2',
                914: '31d6cfe0',
              }[e] +
              '.css'
            );
          }),
          (u.g = (function () {
            if ('object' == typeof globalThis) return globalThis;
            try {
              return this || new Function('return this')();
            } catch (e) {
              if ('object' == typeof window) return window;
            }
          })()),
          (u.hmd = function (e) {
            return (
              (e = Object.create(e)).children || (e.children = []),
              Object.defineProperty(e, 'exports', {
                enumerable: !0,
                set: function () {
                  throw new Error(
                    'ES Modules may not assign module.exports or exports.*, Use ESM export syntax, instead: ' +
                      e.id,
                  );
                },
              }),
              e
            );
          }),
          (u.o = function (e, t) {
            return Object.prototype.hasOwnProperty.call(e, t);
          }),
          (e = {}),
          (u.l = function (t, n, r) {
            if (e[t]) e[t].push(n);
            else {
              var o, i;
              if (void 0 !== r)
                for (var f = document.getElementsByTagName('script'), a = 0; a < f.length; a++) {
                  var c = f[a];
                  if (c.getAttribute('src') == t) {
                    o = c;
                    break;
                  }
                }
              o ||
                ((i = !0),
                ((o = document.createElement('script')).charset = 'utf-8'),
                (o.timeout = 120),
                u.nc && o.setAttribute('nonce', u.nc),
                (o.src = t)),
                (e[t] = [n]);
              var d = function (n, r) {
                  (o.onerror = o.onload = null), clearTimeout(l);
                  var u = e[t];
                  if (
                    (delete e[t],
                    o.parentNode && o.parentNode.removeChild(o),
                    u &&
                      u.forEach(function (e) {
                        return e(r);
                      }),
                    n)
                  )
                    return n(r);
                },
                l = setTimeout(d.bind(null, void 0, { type: 'timeout', target: o }), 12e4);
              (o.onerror = d.bind(null, o.onerror)),
                (o.onload = d.bind(null, o.onload)),
                i && document.head.appendChild(o);
            }
          }),
          (u.r = function (e) {
            'undefined' != typeof Symbol &&
              Symbol.toStringTag &&
              Object.defineProperty(e, Symbol.toStringTag, { value: 'Module' }),
              Object.defineProperty(e, '__esModule', { value: !0 });
          }),
          (u.nmd = function (e) {
            return (e.paths = []), e.children || (e.children = []), e;
          }),
          (u.p = '/'),
          (t = function (e) {
            return new Promise(function (t, n) {
              var r = u.miniCssF(e),
                o = u.p + r;
              if (
                (function (e, t) {
                  for (var n = document.getElementsByTagName('link'), r = 0; r < n.length; r++) {
                    var o = (i = n[r]).getAttribute('data-href') || i.getAttribute('href');
                    if ('stylesheet' === i.rel && (o === e || o === t)) return i;
                  }
                  var u = document.getElementsByTagName('style');
                  for (r = 0; r < u.length; r++) {
                    var i;
                    if ((o = (i = u[r]).getAttribute('data-href')) === e || o === t) return i;
                  }
                })(r, o)
              )
                return t();
              !(function (e, t, n, r) {
                var o = document.createElement('link');
                (o.rel = 'stylesheet'),
                  (o.type = 'text/css'),
                  (o.onload = n),
                  (o.onerror = function (n) {
                    var u = (n && n.target && n.target.href) || t,
                      i = new Error('Loading CSS chunk ' + e + ' failed.\n(' + u + ')');
                    (i.code = 'CSS_CHUNK_LOAD_FAILED'),
                      (i.request = u),
                      o.parentNode.removeChild(o),
                      r(i);
                  }),
                  (o.href = t),
                  document.head.appendChild(o);
              })(e, o, t, n);
            });
          }),
          (n = { 666: 0 }),
          (u.f.miniCss = function (e, r) {
            n[e]
              ? r.push(n[e])
              : 0 !== n[e] &&
                { 76: 1, 121: 1, 261: 1, 365: 1, 481: 1, 536: 1, 874: 1 }[e] &&
                r.push(
                  (n[e] = t(e).then(
                    function () {
                      n[e] = 0;
                    },
                    function (t) {
                      throw (delete n[e], t);
                    },
                  )),
                );
          }),
          (function () {
            var e = { 666: 0 },
              t = [];
            u.f.j = function (t, n) {
              var r = u.o(e, t) ? e[t] : void 0;
              if (0 !== r)
                if (r) n.push(r[2]);
                else {
                  var o = new Promise(function (n, o) {
                    r = e[t] = [n, o];
                  });
                  n.push((r[2] = o));
                  var i = u.p + u.u(t),
                    f = new Error();
                  u.l(
                    i,
                    function (n) {
                      if (u.o(e, t) && (0 !== (r = e[t]) && (e[t] = void 0), r)) {
                        var o = n && ('load' === n.type ? 'missing' : n.type),
                          i = n && n.target && n.target.src;
                        (f.message = 'Loading chunk ' + t + ' failed.\n(' + o + ': ' + i + ')'),
                          (f.name = 'ChunkLoadError'),
                          (f.type = o),
                          (f.request = i),
                          r[1](f);
                      }
                    },
                    'chunk-' + t,
                  );
                }
            };
            var n = function () {};
            function r() {
              for (var n, r = 0; r < t.length; r++) {
                for (var o = t[r], i = !0, f = 1; f < o.length; f++) {
                  var a = o[f];
                  0 !== e[a] && (i = !1);
                }
                i && (t.splice(r--, 1), (n = u((u.s = o[0]))));
              }
              return 0 === t.length && (u.x(), (u.x = function () {})), n;
            }
            u.x = function () {
              (u.x = function () {}), (i = i.slice());
              for (var e = 0; e < i.length; e++) o(i[e]);
              return (n = r)();
            };
            var o = function (r) {
                for (var o, i, [a, c, d, l] = r, s = 0, p = []; s < a.length; s++)
                  (i = a[s]), u.o(e, i) && e[i] && p.push(e[i][0]), (e[i] = 0);
                for (o in c) u.o(c, o) && (u.m[o] = c[o]);
                for (d && d(u), f(r); p.length; ) p.shift()();
                return l && t.push.apply(t, l), n();
              },
              i = (self.webpackChunk = self.webpackChunk || []),
              f = i.push.bind(i);
            i.push = o;
          })(),
          u.x();
      })();
      //# sourceMappingURL=runtime.1d37bf73.js.map
    </script>
    <script src="/304.d691046d.js" defer="defer"></script>
    <script src="/app.5ebfa832.js" defer="defer"></script>
  </body>
</html>

```

### CODE REVIEW

File: website/src/index.html [hunk 1]
```diff
@@ -9,7 +9,15 @@
     <link href="<%= htmlWebpackPlugin.files.publicPath %>safari-pinned-tab.svg" rel="mask-icon" color="#ff5138">
     <link href="<%= htmlWebpackPlugin.files.publicPath %>manifest.json" rel="manifest">
     <link href="<%= htmlWebpackPlugin.options.moduleListUrl %>" rel="preload" as="fetch" crossorigin="anonymous">
-    <link href="<%= htmlWebpackPlugin.options.venuesUrl %>" rel="prefetch" />
+    <link href="<%= htmlWebpackPlugin.options.venuesUrl %>" rel="prefetch">
+    <% const loadedJsFiles = htmlWebpackPlugin.files.js.filter((file) => !file.includes("runtime")); %>
+    <% const inlinedJsFiles = htmlWebpackPlugin.files.js.filter((file) => file.includes("runtime")); %>
```
Comment (ZhangYiJiang) [inline website/src/index.html:14]: Probably better to lift this logic to the webpack config instead and pass them down via `htmlWebpackPlugin`. Also use `_.partition`

**Review by taneliang — COMMENTED** [review]
Love it 😍! I didn't think of this approach.

Would this also require changes to the timetable-only config?

File: website/src/index.ejs [hunk 2]
```diff
@@ -8,33 +8,33 @@
...
+    <link href="<%= htmlWebpackPlugin.files.publicPath %>opensearch.xml" rel="search" title="<%= brandName %> Modules" type="application/opensearchdescription+xml">
     <meta name="theme-color" content="#ffffff" />
   </head>
   <body>
     <div id="app"></div>
-    <noscript>Please enable JavaScript to use <%= htmlWebpackPlugin.options.brandName %></noscript>
+    <noscript>Please enable JavaScript to use <%= brandName %></noscript>
 
-    <script src="/<%= htmlWebpackPlugin.options.browserWarningPath %>" async defer></script>
+    <% if (typeof browserWarningPath !== 'undefined') { %>
```
Comment (taneliang) [inline website/src/index.ejs:35]: Added this check to fix a dev-only runtime error: the line below used to load "/" (i.e. index.html) as a script when `browserWarningPath` was empty. This error was present on master.

File: website/webpack/webpack.config.dev.js [hunk 3]
```diff
@@ -34,12 +38,38 @@ const developmentConfig = merge([
     target: 'web',
     plugins: [
       new HtmlWebpackPlugin({
-        template: path.join(parts.PATHS.src, 'index.html'),
+        template: path.join(parts.PATHS.src, 'index.ejs'),
```
Comment (taneliang) [inline website/webpack/webpack.config.dev.js:41]: Renamed index.html -> index.ejs

**Review by taneliang — APPROVED** [review]
I've fixed this up. @li-kai do you want to take a look at my changes?

**Review by li-kai — COMMENTED** [review]
LGTM. Apologies for leaving this PR to rot. :shipit:

**Pull request merged**